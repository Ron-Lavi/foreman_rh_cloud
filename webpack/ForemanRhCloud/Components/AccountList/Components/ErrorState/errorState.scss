$error_color: #d10d0d;

.error_state {
  margin-top: 100px;
  text-align: center;
  font-weight: bold;
  font-size: 16px;

  .error_icon {
    color: $error_color;
    margin-right: 15px;
    margin-bottom: 10px;
  }

  .error_description {
    color: $error_color;
  }
}
