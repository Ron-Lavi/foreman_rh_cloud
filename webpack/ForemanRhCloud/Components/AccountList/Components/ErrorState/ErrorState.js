import React from 'react';
import PropTypes from 'prop-types';
import { Icon } from 'patternfly-react';
import { translate as __ } from 'foremanReact/common/I18n';
import './errorState.scss';

const ErrorState = ({ error }) => (
  <div className="error_state">
    <Icon className="error_icon" name="times" size="2x" />
    <p>{__('Encountered an error while trying to access the server:')}</p>
    <p className="error_description">{error}</p>
  </div>
);

ErrorState.propTypes = {
  error: PropTypes.string,
};

ErrorState.defaultProps = {
  error: '',
};

export default ErrorState;
