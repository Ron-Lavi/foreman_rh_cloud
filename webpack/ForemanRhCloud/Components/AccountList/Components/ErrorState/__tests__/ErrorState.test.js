import { testComponentSnapshotsWithFixtures } from '@theforeman/test';

import ErrorState from '../ErrorState';

const fixtures = {
  'render without Props': {},
  /** fixtures, props for the component */
};

describe('ErrorState', () => {
  describe('rendering', () =>
    testComponentSnapshotsWithFixtures(ErrorState, fixtures));
});
