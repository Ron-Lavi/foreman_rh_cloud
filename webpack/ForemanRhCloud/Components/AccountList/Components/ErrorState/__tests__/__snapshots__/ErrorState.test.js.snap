// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorState rendering render without Props 1`] = `
<div
  className="error_state"
>
  <Icon
    className="error_icon"
    name="times"
    size="2x"
    type="fa"
  />
  <p>
    Encountered an error while trying to access the server:
  </p>
  <p
    className="error_description"
  />
</div>
`;
